low-level/files/opt/seagate/sspl/setup/sspl_config.py

```diff
@@ -0,0 +1,302 @@
+                        "Run this command with root privileges!!")
+
+        if not os.path.isfile(consts.file_store_config_path):
+            raise SetupError(
+                        errno.EINVAL,
+                        "Missing configuration!! Create and rerun.",
+                        consts.file_store_config_path)
+
+        # Put minion id, consul_host and consul_port in conf file
+        # Onward LDR_R2, salt will be abstracted out and it won't
+        # exist as a hard dependeny of SSPL
+        if consts.PRODUCT_NAME == "LDR_R1":
+            from cortx.sspl.bin.salt_util import SaltInterface
+            salt_util = SaltInterface()
+            salt_util.update_config_file(consts.file_store_config_path)
+
+        if os.path.isfile(self.SSPL_CONFIGURED):
+            os.remove(self.SSPL_CONFIGURED)
+
+        # validate Product
+        product = Conf.get('global_config', 'release>product')
+
+        if not consts.enabled_products:
+            raise SetupError(errno.EINVAL, "No enabled products!")
+        
+        if product not in consts.enabled_products:
+            raise SetupError(
+                        errno.EINVAL,
+                        "Product '%s' is not in enabled products list: %s",
+                        product, consts.enabled_products)
+
+        # Add sspl-ll user to required groups and sudoers file etc.
+        sspl_reinit = [f"{self.DIR_NAME}/bin/sspl_reinit", product]
+        _ , error, returncode = SimpleProcess(sspl_reinit).run()
+        if returncode:
+            raise SetupError(returncode,
+                    "%s/bin/sspl_reinit failed for product %s with error : %e",
+                      self.DIR_NAME, product, error
+                    )
+
+        os.makedirs(self.SSPL_CONFIGURED_DIR, exist_ok=True)
+        with open(self.SSPL_CONFIGURED, 'a'):
+            os.utime(self.SSPL_CONFIGURED)
+
+        # SSPL Log file configuration
+        # SSPL_LOG_FILE_PATH = self.getval_from_ssplconf('sspl_log_file_path')
+        SSPL_LOG_FILE_PATH = Conf.get('sspl', 'SYSTEM_INFORMATION>sspl_log_file_path')
+        if SSPL_LOG_FILE_PATH:
+            self.replace_expr(self.RSYSLOG_SSPL_CONF,
+                        'File.*[=,"]', 'File="%s"' % SSPL_LOG_FILE_PATH)
+            self.replace_expr(
+                    f"{self.DIR_NAME}/low-level/files/etc/logrotate.d/sspl_logs",
+                    0, SSPL_LOG_FILE_PATH)
+
+        # IEM configuration
+        # Configure log file path in Rsyslog and logrotate configuration file
+        # LOG_FILE_PATH = self.getval_from_ssplconf('log_file_path')
+        LOG_FILE_PATH = Conf.get('sspl', 'SYSTEM_INFORMATION>log_file_path')
+
+        if LOG_FILE_PATH:
+            self.replace_expr(self.RSYSLOG_CONF,
+                    'File.*[=,"]', 'File="%s"' % LOG_FILE_PATH)
+            self.replace_expr(
+                    f'{self.DIR_NAME}/low-level/files/etc/logrotate.d/iem_messages',
+                    0, LOG_FILE_PATH)
```

Comment:
Can you add comment here, what this is doing ?

Reply:
# Configure log file path in Rsyslog and logrotate configuration file 
this comment is present @sandeepsgt , should I add function description in replace_expr??